test_iokit.py

```diff
@@ -122,3 +159,206 @@ def test_IOObjectCopySuperclassForClass():
+        )
+    assert err == 0
+    assert iokit.IOIteratorIsValid(iterator) == True
+
+
+def test_IORegistryEntryFromPath():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/IOResources".encode()
+    )
+    assert device != iokit.NULL
+
+    err = iokit.IOObjectRelease(device)
+    assert err == 0
+
+
+def test_IORegistryEntryGetLocationInPlane():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/PEGP@0".encode()
+    )
+    assert device != iokit.NULL
+
+    err, location = iokit.IORegistryEntryGetLocationInPlane (
+        device,
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert location.replace(b"\x00", b"") == b"0"
+
+
+def test_IORegistryEntryGetPath():
+    device = _get_IOResources()
+
+    err, path = iokit.IORegistryEntryGetPath (
+        device,
+        "IOService".encode(),
+        None
+    )
+    assert err == 0
+    assert path.replace(b"\x00", b"") == b"IOService:/IOResources"
+
+
+def test_IORegistryEntryCopyPath():
+    device = _get_IOResources()
+
+    path = iokit.IORegistryEntryCopyPath(device, "IOService".encode())
+    assert path == "IOService:/IOResources"
+
+
+def test_IORegistryEntryIDMatching():
+    device = iokit.IORegistryEntryIDMatching(4294967578)
+    assert iokit.corefoundation_to_native (device) == {
+        "IORegistryEntryID": 4294967578
+    }
+
+
+def test_IORegistryEntryGetRegistryEntryID():
+    device = _get_IOResources()
+    
+    err, obj_id = iokit.IORegistryEntryGetRegistryEntryID (
+            device,
+            None
+        )
+    assert err == 0
+    assert obj_id == 4294967578
+
+
+def test_IORegistryEntryGetChildIterator():
+    device = _get_IOResources()
+    
+    err, iterator = iokit.IORegistryEntryGetChildIterator (
+            device,
+            "IOService".encode(),
+            None
+        )
+    assert err == 0
+
+    interface = list(iokit.ioiterator_to_list(iterator))
+
+    err, props = iokit.IORegistryEntryCreateCFProperties (
+            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
+        )
+    assert err == 0
+    assert props.get("IOProviderClass") == "IOResources"
+
+    for i in interface:
+        err = iokit.IOObjectRelease(i)
+        assert err == 0
+
+
+def test_IORegistryCreateIterator():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IODeviceTree".encode(),
+            iokit.kIORegistryIterateRecursively,
+            None
+        )
+    assert err == 0
+
+    interface = list(iokit.ioiterator_to_list(iterator))
+
+    err, props = iokit.IORegistryEntryCreateCFProperties (
+            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
+        )
+    assert err == 0
+    assert props.get("IOPlatformUUID") == "115C2C75-F539-52BE-90AF-EBBC65ED6CB8"
```

Comment:
```py
FAILED test_iokit.py::test_IORegistryCreateIterator - AssertionError: assert '63890386-B6B...-60DD22AE6A6E' == '115C2C75-F53...-EBBC65ED6CB8'```

Reply:
Could suffice to just check if `props.get("IOPlatformUUID")` is not equal to `None`:
```py
def test_IORegistryCreateIterator():
    err, iterator = iokit.IORegistryCreateIterator (
            iokit.kIOMainPortDefault,
            "IODeviceTree".encode(),
            iokit.kIORegistryIterateRecursively,
            None
        )
    assert err == 0

    interface = list(iokit.ioiterator_to_list(iterator))

    err, props = iokit.IORegistryEntryCreateCFProperties (
            interface[0], None, iokit.kCFAllocatorDefault, iokit.kNilOptions
        )
    assert err == 0
    assert props.get("IOPlatformUUID") != None

    for i in interface:
        err = iokit.IOObjectRelease(i)
        assert err == 0
```

Reply:
sure